eeg_eyetracking_parser/_eeg_preprocessing.py

```diff
@@ -0,0 +1,200 @@
+import numpy as np
+import mne
+import os
+from . import trial_trigger
+from mne.preprocessing import ICA
+import autoreject as ar
+import json
+import matplotlib.pyplot as plt
+
+
+def rereference_channels(raw, ref_channels=['A1', 'A2']):
+    """
+    Rereference EEG data to the mastoids
+    """
+    raw.set_eeg_reference(ref_channels=ref_channels)
+    raw.drop_channels(ref_channels)
+    return raw
+
+
+def drop_unused_channels(raw, chan_name_pattern='Channel'):
+    """
+    Drop unused channels. The pattern is that they all start with "Channel"
+    """
+    empty_chan_names = [chan_name for chan_name in raw.info['ch_names'] if chan_name.startswith(chan_name_pattern)]
+    raw.drop_channels(empty_chan_names)
+    return raw
+
+
+def create_eog_channels(raw, eog_channels=['VEOGT', 'VEOGB', 'HEOGL', 'HEOGR']):
+    """
+    Create EOG channels by subtracting corresponding channels
+    (VEOG = VEOGT - VEOGB)
+    (HEOG = HEOGL - HEOGR)
+    """
+    chinds = [raw.ch_names.index(ichannel) for ichannel in eog_channels]
+    veog = raw._data[chinds[0], :] - raw._data[chinds[1], :]
+    heog = raw._data[chinds[2], :] - raw._data[chinds[3], :]
+    veog = veog[np.newaxis, :]
+    heog = heog[np.newaxis, :]
+    info = mne.create_info(['VEOG', 'HEOG'], raw.info['sfreq'], ['eog', 'eog'])
+    neweog = mne.io.RawArray(np.r_[veog, heog], info)
+    raw.add_channels([neweog], force_update_info=True)
+    raw.set_channel_types(dict(HEOG='eog', VEOG='eog'))
+    raw.drop_channels(eog_channels)
+    return raw
+
+
+def set_montage(raw, montage_name='standard_1020', plot=False):
+    """
+    Set Montage for EEG channel positions
+    Here we use standard_1020 and thus electrode names and positions are according to international 10-20 system
+    """
+    montage = mne.channels.make_standard_montage(montage_name)
+    if plot:
+        montage.plot()
+    if 'Digi' in raw.info['ch_names']:
+        raw.set_channel_types({'Digi': 'stim'})
+    raw.set_montage(montage, match_case=False)
+    return raw
+
+
+def notch_filter(raw, frequencies_remove=(50, 100, 150, 200)):
+    """
+    Removes power line noise generated by EEG system
+    """
+    raw.notch_filter(frequencies_remove)
+    return raw
+
+
+def band_pass_filter(raw, lf=0.1, hf=40, plot=False):
+    """
+    High-pass and low-pass filter for EEG data (band-pass filter)
+    High-pass removes slow drifts
+    Low-pass removes the short-term fluctuations
+    """
+    raw_unfiltered = raw.copy()
+    raw.filter(l_freq=lf, h_freq=hf, picks='all')
+    if plot:
+        for title, data in zip(['Unfiltered', 'Bandpass filtered'], [raw_unfiltered, raw]):
+            fig = data.plot_psd(fmax=250)
+            fig.suptitle(title)
+    return raw
+
+
+def downsample_data(raw, srate=256):
```

Comment:
Event timestamps should also be adjusted. (Due to the way that the EEG and eye-tracking data are merged, we cannot first downsample and then extract the events from the annotations, or at least not easily.)